File: core/repository/dependency_resolution_engine.py
from __future__ import annotations

from typing import Any, Dict, List

from core.repository.runtime_dependency_engine import resolve_runtime_dependencies
from core.parsers.parser_registry import parse_source


def resolve_repository_dependencies(source: str, path: str = "") -> Dict[str, Any]:
    parsed = parse_source(source, path=path) if source else {}
    return resolve_runtime_dependencies(parsed, source)

File: src/repository/dependencyResolutionEngine.ts
/**
 * Converted from Python: core/repository/dependency_resolution_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { resolveRuntimeDependencies } from "./runtimeDependencyEngine.js";
import { parseSource } from "../parsers/parserRegistry.js";

export function resolveRepositoryDependencies(source: any, path: any = ""): any {
  var parsed: any = (py.truthy(source) ? parseSource(source, path) : {});
  return resolveRuntimeDependencies(parsed, source);
}
export { parseSource, resolveRuntimeDependencies };
